board/rt1051/crashdump/crashdumpwriter_vfs.cpp

```diff
@@ -1,31 +1,46 @@
-// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// Copyright (c) 2017-2023, Mudita Sp. z.o.o. All rights reserved.
 // For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
 
 #include "crashdumpwriter_vfs.hpp"
-
+#include <crashdump-serial-number/crashdump_serial_number.hpp>
 #include <cstdio>
-#include <log/log.hpp>
 #include <fcntl.h>
 #include "purefs/vfs_subsystem.hpp"
 #include <purefs/filesystem_paths.hpp>
 #include <exit_backtrace.h>
 
+#include <log/log.hpp>
 #include <filesystem>
 #include <stdint.h>
 #include <unistd.h>
 
+namespace
+{
+    constexpr inline auto suffix = "_crashdump.hex";
+
+    // Crashdump filename pattern:
+    // [serial-number]_[timestamp]_crashdump.hex
+
+    inline std::string generate_crashdump_filename()
+    {
+        time_t crash_time;
```

Comment:
`<ctime>` should be included to use this. Also, `std::chrono::time_point` would be slightly better.